src/libNode/Node.cpp

```diff
@@ -958,35 +989,23 @@ bool Node::StartRetrieveHistory(const SyncType syncType,
         m_mediator.m_ds->m_mapNodeReputation);
   }
 
-  if (REJOIN_NODE_NOT_IN_NETWORK && !LOOKUP_NODE_MODE && !bDS &&
-      !bInShardStructure) {
-    LOG_GENERAL(WARNING,
-                "Node " << m_mediator.m_selfKey.second
-                        << " is not in network, apply re-join process instead");
-    // wait until next two txblocks are mined to give lookup enough time to
-    // upload incr data to S3.
-    unique_lock<mutex> lock(m_mediator.m_lookup->m_MutexCVSetTxBlockFromSeed);
-    m_mediator.m_lookup->SetSyncType(SyncType::RECOVERY_ALL_SYNC);
-
-    uint64_t oldBlkCount = m_mediator.m_txBlockChain.GetBlockCount();
-    LOG_GENERAL(INFO, "Wait until next two txblock are recvd from lookup..");
-    while (true) {
-      do {
-        m_mediator.m_lookup->GetTxBlockFromSeedNodes(
-            m_mediator.m_txBlockChain.GetBlockCount(), 0);
-      } while (m_mediator.m_lookup->cv_setTxBlockFromSeed.wait_for(
-                   lock, chrono::seconds(RECOVERY_SYNC_TIMEOUT)) ==
-               cv_status::timeout);
-
-      if (m_mediator.m_txBlockChain.GetBlockCount() > oldBlkCount + 1) {
-        LOG_GENERAL(INFO, "Received next two txblocks. Ok to rejoin now!")
-        break;
+  if (REJOIN_NODE_NOT_IN_NETWORK && !LOOKUP_NODE_MODE && !bDS) {
+    if (!bInShardStructure) {
+      LOG_GENERAL(
+          WARNING,
+          "Node " << m_mediator.m_selfKey.second
+                  << " is not in network, apply re-join process instead");
+      WaitForNextTwoBlocksBeforeRejoin();
+      return false;
+    } else if (bIpChanged) {
```

Comment:
What happens if `bIpChanged` is false?

Reply:
It will follow usual path i.e. it will join a shard because `bInShardStructure` is true ( IP is same in this case )